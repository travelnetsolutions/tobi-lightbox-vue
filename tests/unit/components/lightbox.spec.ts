import { shallowMount } from '@vue/test-utils';
import Lightbox from '@/components/lightbox.vue';
import Slide from '@/components/slide.vue';
import { gallery } from './fixtures/galleryFixture';

describe('Lightbox.vue', () => {

  let props: {};

  const build = () => {
    const wrapper = shallowMount(Lightbox, {
      propsData: props
    });
    return {
      wrapper,
      slides: () => wrapper.findAll(Slide),
      firstSlide: () => wrapper.find(Slide),
      anySlide: () => wrapper.find(Slide)
    };
  };

  beforeEach(() => {
    props = { images: gallery };
  });

  it('renders the component', () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders child components', () => {
    const { slides } = build();
    expect(slides().exists()).toBe(true);
  });

  it('Has images', () => {
    const { wrapper } = build();
    expect(wrapper.vm.$props.images).toBe(gallery);
  });

  it('doesn\'t render slides with empty images array', () => {
    const { slides, wrapper } = build();
    wrapper.setProps({
      images: [],
    });
    expect(slides().exists()).toBe(false);
  });

  it('Passes an image to the child component', () => {
    const { firstSlide, wrapper } = build();
    expect(firstSlide().props().image.imgsrc).toBe(wrapper.props().images[0].imgsrc);
  });

  it('toggles the lightbox when a slide emits the toggle event', () => {
    const { anySlide, wrapper } = build();
    anySlide().vm.$emit('toggle');
    expect(wrapper.emitted().input).toBeTruthy();
  });
});
