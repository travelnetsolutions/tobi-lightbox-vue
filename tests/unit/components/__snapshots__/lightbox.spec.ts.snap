// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lightbox.vue renders the component 1`] = `
<div role="dialog" aria-hidden="true" class="tobi">
  <div class="tobi__overlay"></div>
  <div class="tobi__slider" style="transform: translate3d(0px, 0, 0);">
    <slide-stub index="0" currentindex="0" image="[object Object]"></slide-stub>
    <slide-stub index="1" currentindex="0" image="[object Object]"></slide-stub>
    <slide-stub index="2" currentindex="0" image="[object Object]"></slide-stub>
    <slide-stub index="3" currentindex="0" image="[object Object]"></slide-stub>
    <slide-stub index="4" currentindex="0" image="[object Object]"></slide-stub>
  </div>
  <!----> <button type="button" aria-label="Next" class="tobi__next"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="1" stroke-linecap="round" stroke-linejoin="round">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg></button> <button type="button" aria-label="Close" class="tobi__close"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
      <line x1="18" y1="6" x2="6" y2="18"></line>
      <line x1="6" y1="6" x2="18" y2="18"></line>
    </svg></button>
  <div class="tobi__counter">1/5</div>
</div>
`;
